### ISSUE TITLE
Webaudio example gives error in Safari on Mac (works in Firefox)

### ISSUE THREAD

**Issue by kud1ing** [rustwasm/wasm-bindgen#897]
The web development console says:
```
[Error] Unhandled Promise Rejection: ReferenceError: Can't find variable: AudioContext
	(anonymous function)
	promiseReactionJob
```

This is Safari 12 on macOS Mojave.

**Comment by chinedufn** [comment]
Web audio is [supported with a prefix in safari](https://caniuse.com/#feat=audio-api), so the example [would need to fall back to `webkitAudioContext` if there was no `AudioContext`](https://github.com/rustwasm/wasm-bindgen/blob/826619870f4cb043663530c8c6a60e486aa5b2b7/examples/webaudio/src/lib.rs#L6)

---

From a quick search it doesn't seem like `web-sys` supports any prefixed APIs (I could very well be missing something though!). 

If that's the case I'm also not sure if that's an intentional decision or not.

**Comment by alexcrichton** [comment]
Thanks for the report and thanks for the info @chinedufn! We [discussed this during today's meeting](https://paper.dropbox.com/doc/RustWebAssembly-WG--AN0Kj_TSmeR5csz9JFryWJQsAg-dcXi5F7Bqlvg10zd7dKve#:uid=466221811040185070621428&h2=2018/09/27) and the conclusion was that we'd like to add support to `web-sys` to whitelist APIs as "these are alternative class names". That way we can get this working by default even in Safari where it's renamed to `webkitAudioContext`.

**Comment by alexcrichton** [comment]
https://github.com/rustwasm/wasm-bindgen/pull/906 fixes using `webkitAudioContext` on Safari, but unfortunately the example still doesn't work in Safari. The `AudioScheduledSourceNode` type doesn't exist in Safari (and edge presumably) but rather the methods are just listed in the parent classes. I've written up a more detailed issue about that at https://github.com/rustwasm/wasm-bindgen/issues/908

**Comment by alexcrichton** [comment]
The combination of https://github.com/rustwasm/wasm-bindgen/pull/909 and https://github.com/rustwasm/wasm-bindgen/pull/906 should fix this issue

**Comment by kud1ing** [comment]
Is this deployed? The demo now gives:

```
ReferenceError: Can't find variable: AudioScheduledSourceNode
Eval Code
eval
./webaudio.js — 0.index.js:11
__webpack_require__ — index.js:144
__webpack_require__
promiseReactionJob
```

**Comment by alexcrichton** [comment]
Ok I've confirmed the deployed example does indeed work in safari now!

**Comment by kud1ing** [comment]
Me too, thanks.

**Comment by tat-m3dicine** [comment]
Do you guys have in mind for webkitOfflineAudioContext ? Currently `new_with_context_options` and `new_with_number_of_channels_and_length_and_sample_rate` returns `OfflineAudioContext` which safari doesn't support

**Comment by alexcrichton** [comment]
@tat-m3dicine is that perhaps another case that's [needed here](https://github.com/rustwasm/wasm-bindgen/blob/aca49e1a6e1e29b6ffe8430b5fe83eaabc9c9dc7/crates/webidl/src/lib.rs#L569-L577)? Would that fix the issue for you?

**Comment by tat-m3dicine** [comment]
Im not familiar the usage with webidl directly or what not. From that segment of codes, I think it's missing `OfflineAudioContext` match arm for vendor-prefix. 

Bit of relief, `AudioContext` works as intended in `pkg/<project_name>.js`

**Comment by tat-m3dicine** [comment]
@alexcrichton  Indeed, it's missing another case. Atm, I will just deal with OfflineAudioContext using `extern "C" {...}`. Do you guys have in mind when will be the release date for this update?

**Comment by alexcrichton** [comment]
I've created https://github.com/rustwasm/wasm-bindgen/pull/1865 for this, @tat-m3dicine mind trying that out and seeing if it works for you?
